pkg/engine/update.go

```diff
@@ -667,9 +658,8 @@ type previewActions struct {
 	MapLock sync.Mutex
 }
 
-func shouldReportStep(step deploy.Step, opts *deploymentOptions) bool {
-	return step.Op() != deploy.OpRemovePendingReplace &&
-		(opts.reportDefaultProviderSteps || !isDefaultProviderStep(step))
+func shouldReportStep(step deploy.Step) bool {
```

Comment:
Just bikeshedding here, so feel free to ignore.  I'm wondering if we should rename this to something like `isInternalStep`, since reporting the step is just a consequence of that.